x-pack/plugins/rule_registry/server/utils/persistence_types.ts

```diff
@@ -27,7 +29,9 @@ export type PersistenceAlertService = <T>(
 ) => Promise<PersistenceAlertServiceResult<T>>;
 
 export interface PersistenceAlertServiceResult<T> {
-  createdAlerts: Array<T & { _id: string; _index: string }>;
+  createdAlerts: Array<
+    AlertWithCommonFieldsLatest<T> & { _id: string; _index: string; [key: string]: SearchTypes }
```

Comment:
So what's happening here is the `PersistenceAlertService` takes in documents of type `T` and writes them to Elasticsearch after injecting the common fields into each document - making them `AlertWithCommonFieldsLatest<T>`. `AlertWithCommonFields<T>` is intended to be the type of the `_source` of the written documents. When the function returns though, it also includes the `_id` and `_index` metadata that Elasticsearch returns so that we can make those values available in the actions context for the Cases connector, even though those fields aren't part of `_source`. 

I think you're right that `[key: string]: SearchTypes` index signature is unnecessary though so I'll remove it.